llvm/lib/Target/PowerPC/PPCMergeStringPool.cpp

```diff
@@ -215,6 +215,11 @@ void PPCMergeStringPool::collectCandidateConstants(Module &M) {
         Global.getLinkage() != GlobalValue::InternalLinkage)
       continue;
 
+    // Do not pool thread-local constants, as the pooled strings can contain
+    // non-thread-local constants, and these should not be mixed together.
+    if (Global.isThreadLocal())
+      continue;
+
```

Comment:
Can we combine this conditon to the top exit condtion at line 174?  So that would change the doc on line 173 to:
```
// We can only pool non-thread-local constants.
```